From 44fc85c6f1152e530eb93700feb4b0732f2f1503 Mon Sep 17 00:00:00 2001
From: "Andrew F. Davis" <afd@ti.com>
Date: Thu, 31 Mar 2016 00:32:48 +0000
Subject: [PATCH 43/46] TEMP: net: prueth: Reduce error console message noise

A full transmit queue is not a serious error and occurs often in high
traffic situations so informing the user at each occurrence creates
unneeded console noise. Do not print an error message for this type of
transmit failure.

Signed-off-by: Andrew F. Davis <afd@ti.com>
Tested-by: Jason Reeder <jreeder@ti.com>
---
 drivers/net/ethernet/ti/ti-prueth.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/ti/ti-prueth.c b/drivers/net/ethernet/ti/ti-prueth.c
index cb927d4..c84b3f7 100644
--- a/drivers/net/ethernet/ti/ti-prueth.c
+++ b/drivers/net/ethernet/ti/ti-prueth.c
@@ -1230,8 +1230,8 @@ static int emac_ndo_start_xmit(struct sk_buff *skb, struct net_device *ndev)
 	}
 
 	if (ret) {
-		if (netif_msg_tx_err(emac) && net_ratelimit())
-			netdev_err(ndev, "packet queue failed: %d", ret);
+		if (ret != -ENOBUFS && netif_msg_tx_err(emac) && net_ratelimit())
+			netdev_err(ndev, "packet queue failed: %d\n", ret);
 		goto fail_tx;
 	}
 
-- 
1.9.1

